Sources/Fuzzilli/Minimization/Verifier.swift

```diff
@@ -47,7 +47,7 @@ class ReductionVerifier {
         }
 
         // At this point, the code must be statically valid though.
-        assert(code.isStaticallyValid())
+        guard code.isStaticallyValid() else { return false }
```

Comment:
Why this change?

Reply:
Block reductions can lead to statically invalid code so rather that triggering asserts and failing silently on release builds I thought it makes sense to convert the assert to guard. For example consider the following async function that's being reduced:
```
BeginAsyncFunc
....
Await v0
...
EndAsyncFunc
```
When this block is reduced the reducer first tries to Nop out the Being and End instructions and then test the remaining instructions for static validity. This worked before we introduced context analysis to `Code.check` but now the check would fail due to `Await` being introduced to a non-default context.

So it made sense to me to have the assertion converted to a guard since if we don't have a statically valid program at that point of execution then we bail out of the reduction.

Reply:
I see, thanks for explaining! I'm a bit worried that this will potentially hide bugs in reducers. Let's say a reducer generates statically invalid code due to a bug, we'd then never find out because it will just return here instead of failing.
On the other hand, it is a nice and simple solution. 
So how about just adding a comment here: 
```
TODO returning silently here might hide bugs in Reducers. However, certain Reducers are currently
allowed to generate statically invalid programs, for example the BlockReducer. To fix that, we should
change the return type of this function to an enum and encode the reason for failure. That way, reducers
that are expected to always generate statically valid programs can simply assert that that's the case.
```

When we do this, do we still need the changes in BlockReducer.swift?

Reply:
> When we do this, do we still need the changes in BlockReducer.swift?

The changes in BlockReducer were mainly to optimise this reduction logic to save us a few cpu cycles. I can revert those changes.